src/http/SOCKSProxy.zig

```diff
@@ -0,0 +1,240 @@
+                return error.UnsupportedSOCKSVersion;
+            }
+
+            if (method == @intFromEnum(SOCKSAuthMethod.no_acceptable)) {
+                return error.SOCKSAuthenticationFailed;
+            }
+
+            if (method == @intFromEnum(SOCKSAuthMethod.no_auth)) {
+                this.state = .auth_complete;
+                try this.sendConnectRequest(is_ssl, socket);
+            } else {
+                return error.UnsupportedSOCKSAuthMethod;
+            }
+
+            return true; // Data was consumed by SOCKS handshake
+        },
+        .connect_request => {
+            if (data.len < 4) {
+                return error.IncompleteSOCKSResponse;
+            }
+
+            const version = data[0];
+            const reply = data[1];
+            // data[2] is reserved
+            const atyp = data[3];
+
+            if (version != @intFromEnum(SOCKSVersion.v5)) {
+                return error.UnsupportedSOCKSVersion;
+            }
+
+            if (reply != @intFromEnum(SOCKSReply.succeeded)) {
+                return error.SOCKSConnectionFailed;
+            }
+
+            // Parse the bound address (we don't need it, but need to skip it)
+            var offset: usize = 4;
+            switch (atyp) {
+                @intFromEnum(SOCKSAddressType.ipv4) => offset += 4,
+                @intFromEnum(SOCKSAddressType.ipv6) => offset += 16,
+                @intFromEnum(SOCKSAddressType.domain_name) => {
+                    if (data.len <= offset) return error.IncompleteSOCKSResponse;
+                    offset += 1 + data[offset]; // domain length + domain
+                },
+                else => return error.UnsupportedSOCKSAddressType,
+            }
+            offset += 2; // port
+
+            if (data.len < offset) {
+                return error.IncompleteSOCKSResponse;
+            }
+
+            this.state = .connected;
+            log("SOCKS proxy connected successfully", .{});
+
+            // SOCKS handshake complete, HTTP traffic can now flow through the tunnel
+            // Don't change proxy_tunneling flag - let the normal flow handle it
+
+            // If there's any remaining data after the SOCKS response, process it as HTTP
+            if (data.len > offset) {
+                return false; // Let HTTP client process remaining data
+            }
+
+            return true; // Data was consumed by SOCKS handshake
+        },
+        .connected => {
+            // Pass through data to the HTTP client
+            return false; // Let HTTP client handle this data
+        },
+        else => {
+            return error.UnexpectedSOCKSState;
+        },
+    }
+}
+
+pub fn close(this: *SOCKSProxy) void {
+    this.state = .failed;
+}
+
+pub fn shutdown(this: *SOCKSProxy) void {
+    this.close();
+}
+
+pub fn detachAndDeref(this: *SOCKSProxy) void {
+    this.deref();
+}
+
+fn deinit(this: *SOCKSProxy) void {
+    bun.destroy(this);
```

Comment:
how is the memory fo rthe destination_port, destination_host, and proxy_url managed? what ensures those values are alive? clone it in the `fn init` and free it in the `deinit`.